### DISCUSSION TITLE
Double click for always front

### DISCUSSION THREAD

**Discussion by BobbyatGithub** [BLumia/pineapple-pictures#38] [Ideas]
Hi, thanks for this extraordinary work! I think it should be practical to add always front to the double click function because it is pretty frequent to switch this status.

**Comment by BLumia** [comment]
Hi and thanks for the feedback! Double-click to toggle Stay-on-Top is not hard to implement, but I'd like to hear about its use case.

Did you use it to put reference images onto other applications and need to toggle that state very frequently? Could you share the detail of the use case so I could know if there is a better approach? e.g. If I introduce a way to bind custom shortcuts to any action(include the "stay-on-top" toggle), will that fulfilled your use case?

**Reply by BobbyatGithub** [reply]
Thank you for your concern! I use Pineapple to put the picture in front of the Bloomberg window to do analysis，I need to cancel the Stay-on-top when zooming in the candle graph. Although I have two screens, I don’t bother to turn my head, lol. In this case, shortcuts may conflict with Bloomberg, but I think custom should work.

**Reply by BLumia** [reply]
Thanks for the info, I'll let you know by replying to this thread if the related feature were added to the application <3

**Reply by BobbyatGithub** [reply]
Another idea is coming! Probably, there should be an option for opening an image, in default,  in 1:1 size with the window adapted rather than maximized. Or open with the window size as it closed last time. Appreciate it!

**Reply by BLumia** [reply]
Adding an option for selecting an image to load seems to be a better approach IMO. btw you can open another discussion thread so we could track different feature requests in their own discussion thread ;)

**Reply by BLumia** [reply]
An update about the manually open image file feature, it should now live in the `master` branch with commit f202bb58f4. I'm not sure which OS/distro you're using, if you are using Windows, then you could download a CI snapshot build at [here](https://ci.appveyor.com/project/BLumia/pineapplepictures/builds/41668254/artifacts) (be sure to download it before it expires).

The new action can be found in the context menu names "Open...", it's not yet translated if you are using a non-English language environment. I'm still not decided on the final copywriting and the behavior so any feedback is welcome. Be sure to create another discussion thread or issue for that feature since that feature is not related to the feature related to this discussion thread (double-click for always front).

**Reply by BLumia** [reply]
> The new action can be found in the context menu names "Open...", it's not yet translated if you are using a non-English language environment. I'm still not decided on the final copywriting and the behavior so any feedback is welcome.

I still think the copywriting "Open..." may cause confusion for users so I decided to remove it from the context menu until a better idea comes out (shortcut <kbd>Ctrl + O</kbd> for *Open* will continue working). If you'd like to use a version that contains that action in the context menu, consider downloading the CI build from the link above before it expires :)
